src/ui/map.js

```diff
@@ -215,6 +218,8 @@ const defaultOptions = {
  * @param {boolean} [options.collectResourceTiming=false] If `true`, Resource Timing API information will be collected for requests made by GeoJSON and Vector Tile web workers (this information is normally inaccessible from the main Javascript thread). Information will be returned in a `resourceTiming` property of relevant `data` events.
  * @param {number} [options.fadeDuration=300] Controls the duration of the fade-in/fade-out animation for label collisions, in milliseconds. This setting affects all symbol layers. This setting does not affect the duration of runtime styling transitions or raster tile cross-fading.
  * @param {boolean} [options.crossSourceCollisions=true] If `true`, symbols from multiple sources can collide with each other during collision detection. If `false`, collision detection is run separately for the symbols in each source.
+ * @param {string} [options.uiLanguage='en'] A language to use in the UI. The only supported languages for now are 'en' & 'uk'. If an unsupported language is provided, 'en' will be used.
+ * @param {object} [options.uiLanguagePatch] A patch to apply to the localization table. You can replace all UI strings (thereby adding support for a new translation) or just part of them (thereby patching the default table for `uiLanguage`).
```

Comment:
These two could be combined. If an object is passed to uiLanguage it acts as the language string lookup, or if just a string it's the language code.

On the other hand you could say how it is now is clearer :thought_balloon: . Interested to see what other people think.

Reply:
@andrewharvey Yes, you are right having it as one of a string or an object might be an option as well. The reasoning behind the route I've taken was that it allows the following scenarios:

1) Only change the default translation for some strings. So, say i specify Swedish as a language and then only "patch" the "Exit fullcsreen" tooltip. If I had to provide the entire list of strings, that would make my code more brittle and susceptible to bugs when new controls/strings added or existing default text is changed. With this "patch" approach i can either change all the strings (thus, providing support for a completely new language/translation) or only few of them.

2) By allowing to "patch" the dictionary, it is easier to add localization support for custom controls. So, ideally, when developing a custom control, one would expose language "packs" for it (and perform the actual translation through this._map._translateUiString('') . Later, a consumer of such control could pass that language "pack" as "uiLanguagePatch" objects. Moreover, if you consume several such controls, you may merge their translation packs, while preserving translation for the default controls.

So, this was my reasoning. May be it is too complicated and a bit simpler approach has to be taken. For example, we could have a central 'localization' repository + some API to patch it. Then, custom controls will have a chance to install their supported languages.

I will wait for some more feedback.

Reply:
I agree with @andrewharvey that it would be nice to keep this a single option, but I also agree with the rationale for 1&2 here. 

What if we had a single option, which always expects an object with a `language` string and optionally a `strings` object with the string mappings? Defaulting to `{ language: 'en' }`

In that case I could pass `{ language: 'uk' }` to use the built-in Ukrainian translations, without any overrides, or something like this to provide a partial or full translation patch for a given language:
```
{ language: 'en', strings: { 'FullscreenControl.ExitTooltip': 'Minimize', ... } }
```

In that case perhaps renaming the option `uiLocalization` or something would avoid the awkwardly repetitive `uiLanguage: { language: ... }`.

Thoughts?